retry_test.go

```diff
@@ -260,3 +260,16 @@ func TestCombineDelay(t *testing.T) {
 		)
 	}
 }
+
+func TestContext(t *testing.T) {
+	start := time.Now()
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+	err := Do(
+		func() error { return errors.New("test") },
+		Context(ctx),
+	)
+	dur := time.Since(start)
+	assert.Error(t, err)
+	assert.True(t, dur < time.Millisecond, "immediately cancellation")
```

Comment:
I guess that anything less than the DefaultDelay might better express the expectation. We might also test that the function is called at most once.

Reply:
Thanks to test hints. This implementation doesn't work well. Retry is called many times and return an error. I think about what would return this code? error? or nothing?

Reply:
I fixed it now - I think...